comp/logs/agent/config/config.go

```diff
@@ -78,16 +77,7 @@ var (
 // GlobalProcessingRules returns the global processing rules to apply to all logs.
 func GlobalProcessingRules(coreConfig pkgconfigmodel.Reader) ([]*ProcessingRule, error) {
 	var rules []*ProcessingRule
-	var err error
-	raw := coreConfig.Get("logs_config.processing_rules")
-	if raw == nil {
-		return rules, nil
-	}
-	if s, ok := raw.(string); ok && s != "" {
-		err = json.Unmarshal([]byte(s), &rules)
-	} else {
-		err = structure.UnmarshalKey(coreConfig, "logs_config.processing_rules", &rules, structure.ConvertEmptyStringToNil)
```

Comment:
We don't need `ConvertEmptyStringToNil` anymore ?

Reply:
Nope, it's made unnecessary by the new logic.